src/testing/models.ts

```diff
@@ -20,21 +20,59 @@ export interface TracerEvent {
   properties: ArbitraryProperties;
 }
 
-export abstract class BaseTestEvaluator<TestCaseType, OutputType> {
+interface TestEvaluator<TestCaseType, OutputType> {
```

Comment:
Thoughts on TestCaseEvaluator?

Reply:
sure but just FYI this is an internal type, not something an end user would use

Reply:
did you mean the evaluator? renaming `BaseTestEvaluator` to `BaseTestCaseEvaluator`?

Reply:
ya i meant also updating `BaseTestEvaluator` to `BaseTestCaseEvaluator`

Reply:
i guess we could but the most accurate name is probably `TestCaseAndItsOutputEvaluator` :p so "test" seemed to capture all of that. we'd have to leave the old name as well for backwards compatibility

Reply:
i'm ok w/ renaming, i think at least `BaseTestCaseEvaluator` is more accurate than `BaseTestEvaluator`. can tackle along with the move to `evaluators` module?

Reply:
Yep that sounds good.